test/known_problems/should_fail/intersection_with_any_fail.erl

```diff
@@ -14,3 +16,26 @@ any_refined_using_guard(X) when is_atom(X) ->
                                                       atom().
 intersection_using_constraits(X) ->
     X.
+
+-spec var_as_pattern(atom()) -> integer().
+var_as_pattern(Atom) ->
+    case get_any() of
+        Atom ->
+	    %% at this point Atom :: any()
+	    %% but we want Atom :: atom() & any()
+            Atom
+    end.
+
+-spec var_inside_pattern(atom()) -> integer().
+var_inside_pattern(Atom) ->
+    case get_any() of
+        {Atom} ->
+            %% at this point Atom :: any()
+            %% but we want Atom :: atom() & any()
+            Atom
+    end.
+
+%% helper
+-spec get_any() -> any().
+get_any() ->
+    receive Any -> Any end.
```

Comment:
Nice tests!

Reply:
I just moved them from another test file. :-) They were failing as they should before but now they are known problems.